devportalservice/devportalservice.go

```diff
@@ -3,43 +3,80 @@ package devportalservice
 import (
 	"bytes"
 	"encoding/json"
-	"errors"
 	"fmt"
 	"io/ioutil"
 	"net/http"
-	"os"
 	"strings"
 	"text/template"
+	"time"
 
 	"github.com/bitrise-io/go-utils/log"
 )
 
-// NetworkError represents a networking issue.
-type NetworkError struct {
-	Status int
-	Body   string
+const (
+	bitriseBuildURLKey      = "BITRISE_BUILD_URL"
+	bitriseBuildAPITokenKey = "BITRISE_BUILD_API_TOKEN"
+)
+
+type httpClient interface {
+	Do(req *http.Request) (*http.Response, error)
 }
 
-func (e NetworkError) Error() string {
-	return fmt.Sprintf("response %d %s", e.Status, e.Body)
+// AppleDeveloperConnectionProvider ...
+type AppleDeveloperConnectionProvider interface {
+	GetAppleDeveloperConnection(buildURL, buildAPIToken string) (*AppleDeveloperConnection, error)
 }
 
-// CIEnvMissingError represents an issue caused by missing environment variables,
-// which environment variables are exposed in builds on Bitrise.io.
-type CIEnvMissingError struct {
-	Key string
+// BitriseClient ...
+type BitriseClient struct {
+	httpClient httpClient
 }
 
-func (e CIEnvMissingError) Error() string {
-	return fmt.Sprintf("%s env is not exported", e.Key)
+// NewBitriseClient ...
+func NewBitriseClient(client httpClient) *BitriseClient {
+	return &BitriseClient{
+		httpClient: client,
+	}
 }
 
-// portalData ...
-type portalData struct {
-	AppleID              string              `json:"apple_id"`
-	Password             string              `json:"password"`
-	ConnectionExpiryDate string              `json:"connection_expiry_date"`
-	SessionCookies       map[string][]cookie `json:"session_cookies"`
+const appleDeveloperConnectionPath = "apple_developer_portal_data.json"
+
+// GetAppleDeveloperConnection ...
+func (c *BitriseClient) GetAppleDeveloperConnection(buildURL, buildAPIToken string) (*AppleDeveloperConnection, error) {
```

Comment:
Is BITRISE_PORTAL_DATA_JSON no longer needed? It seems no longer used.

Reply:
Based on the PR description, setting FASTLANE_SESSION in a previous step replaces this option.

Reply:
For local testing, it's better to expose `FASTLANE_SESSION` directly. Previous `BITRISE_PORTAL_DATA_JSON` stored a whole `AppleDeveloperConnection` JSON object, but only the session cookie was used.
[FASTLANE_SESSION can be generated by](https://docs.fastlane.tools/best-practices/continuous-integration/#two-step-or-two-factor-auth): `fastlane spaceauth -u user@email.com`.